packages/grafana-ui/src/types/data.ts

```diff
@@ -104,4 +104,5 @@ export interface AnnotationEvent {
   text?: string;
   type?: string;
   tags?: string[];
+  source?: any; // source.type === 'dashboard' ????
```

Comment:
Hmm where is this used or what is it used for?

Reply:
ya, that one is weird.  In the existing code, the sources gets added on each query and then checked in:
https://github.com/grafana/grafana/blob/master/public/app/features/annotations/annotations_srv.ts#L46

I'll add a comment in the code.  I think we should rethink this (but not in this PR)